### PR TITLE
Support `swift -e` for immediate execution

### PR DESCRIPTION
**Pull request by keith**
This adds the ability to pass code directly to the Swift interpreter
using `-e`. Usage:

```sh
swift -e 'print("Hello World!")'
```

You can also pass `-e` multiple times:

```sh
swift -e 'print("Hello World!")' -e 'print("Hello again :)")'
```

Resolves [SR-5860](https://bugs.swift.org/browse/SR-5860).

### CODE REVIEW

**Comment by brentdax** [comment]
So, I've actually got a [branch like this from a few weeks ago sitting around](https://github.com/apple/swift/compare/master...brentdax:time-for-some-swift-golf), but there's a major design difference: this branch writes out a temporary file, while mine uses an in-memory file. Is there a reason we should prefer one or the other?

**Comment by keith** [comment]
Ah nice! I'm not sure if there should be a preference here. Hopefully others will weigh in on that. That was the guidance on the ticket, but making the changes you had to make to accept lines or inputs was a little more than I could chew.

File: include/swift/Option/Options.td [hunk 1]
```diff
@@ -677,6 +677,12 @@ def deprecated_integrated_repl : Flag<["-"], "deprecated-integrated-repl">,
   Flags<[FrontendOption, NoBatchOption]>, ModeOpt;
 def i : Flag<["-"], "i">, ModeOpt; // only used to provide diagnostics.
 
+def execute : Separate<["-"], "execute">,
+  HelpText<"Execute the passed Swift code immediately">,
+  Flags<[NoBatchOption]>;
+def e : Separate<["-"], "e">, Alias<execute>,
+  Flags<[NoBatchOption]>, ModeOpt;
```
Comment (jrose-apple) [inline include/swift/Option/Options.td]: I think `ModeOpt` is supposed to go on the non-alias.

**Review by jrose-apple — COMMENTED** [review]
Thanks for picking this up, Keith!

I actually think I prefer the in-memory approach, i.e. building the "source file" on the frontend side rather than the driver side. The reason for this is that what you have here eagerly writes out the temporary file and doesn't have a way to clean it up, which is a problem for, say, the `-###` option (which prints out frontend jobs that are going to execute).

**Comment by keith** [comment]
Sounds good, @brentdax do you want to submit your solution?

**Comment by keith** [comment]
I don't think I'm going to keep going with this one, anyone can feel free to build on this!

**Pull request closed without merging**